providers/linux/os.go

```diff
@@ -68,8 +68,8 @@ func init() {
 	}
 }
 
-func OperatingSystem() (*types.OSInfo, error) {
-	return getOSInfo("")
+func OperatingSystem(procfs string) (*types.OSInfo, error) {
```

Comment:
This is a breaking change to an exported API. Is this change necessary? Hopefully all users are fetching OS metadata through `sysinfo.Host()` -> `Info()` to get the `HostInfo.OS` value.

If we ever make a new major version I want to put providers/ behind internal/.

Reply:
Bah, yeah, I didn't even think of that, I was just considering the `sysinfo.Host()` use case, like you said.

Reply:
Issue for making providers be internal: https://github.com/elastic/go-sysinfo/issues/237